src/widgets/stats/defaultTemplates.js

```diff
@@ -1,5 +1,15 @@
 export default {
```

Comment:
I don't really know how the templates work, but I think we should handle the case where `nbHits === nbSortedHits`, to avoid this kind of sentences `10,000 relevant results sorted out of 10,000 found in 1ms`

Reply:
yes, that's an important note! isSmartSorted should be false in that case IMO

Reply:
I just handled it [like that](https://github.com/algolia/react-instantsearch/pull/3010/files#diff-7f595657982de401bfc86c6843f6e39d019e23ae53ff282e017303121a455a1bR63) in RIS, should I fix it before merging `smartSort`?

Reply:
I think so, but @eunjae-lee and you have been closer to the implementation of this feature, wdyt?

Reply:
If we think `isSmartSorted` reflect what the engine says, it should stay like that. But it make sense to consider that if the number of results are the same we consider it has `not smart sorted`. 🤔

Reply:
Added here: https://github.com/algolia/react-instantsearch/pull/3009/commits/9fa94a9e4274839ae05026581a57d3611a323365, wdyt @eunjae-lee ?

Reply:
I think `isSmartSorted` should stay as is. With `0 < appliedRelevancyStrictness`, it's indeed smartly sorted from the engine. It's just the result happen to be the same as normal sort. Giving another meaning to `isSmartSort` could be confusing (like "why is it false? I set the value on dashboard?")

However it'd be a good addition in detail to render proper text. So I'm thinking of the change only in stats. What do you think? @Haroenv @shortcuts

Reply:
> Giving another meaning to isSmartSort could be confusing (like "why is it false? I set the value on dashboard?")

It could be confusing from a developer POV indeed, but I think it make sense to consider that if the hits are the same as the sorted hits, then it's not smart sorted. As the boolean isn't `isSmartSortConfigured` for example, I think we can take this liberty IMO

Reply:
When it's configured but happens to have the same number of hits, if we follow your logic, we should display a button saying "See relevant results" because isSmartSort === false. And when clicking the button will `refine(undefined)`, which will do nothing (because it's already undefined). That's confusing, no?

Reply:
True, this would add a lot of conditions to add to handle it!

Reply:
Should we disable the button on the default template when the hits are matching then? Or do we let the user handle this case?

Reply:
It will likely happen when there's empty query, etc. IMO we can release the smartSort widget as-is, and update only the default template in stats widget. If we get better understanding and learn patterns, we could add some logic to smartSort, but I feel it's too early to assume conditional behaviors.

Reply:
Ok, I'll revert it to [`nbHits !== nbSortedHits`](https://github.com/algolia/react-instantsearch/pull/3010/files#diff-7f595657982de401bfc86c6843f6e39d019e23ae53ff282e017303121a455a1bR63) then!

Reply:
but let's hear what @Haroenv thinks

Reply:
@Haroenv not sure if I understand you correctly.
My suggestion was

With appliedRelevancyStrictness > 0, when nbSortedHits === nbHits:
1. Display the stat as `1000 hits in 10ms` (without smart sort mentioned)
2. Display the button as `See all results` (still as if it's a smart sort)

Which part do you disagree? Do you think we should display the button as `See relevant results` or not display the button at all?

Reply:
Chiming there! I don't have a *strong opinion* on that topic but I can help by giving you some context and pointers on the feature.

The idea behind the banner was to be "transparent" with the end-users as the algorithm can be perceived as a "black-box". That's why I am not really disturbed by the following:

> `10,000 relevant results sorted out of 10,000 found in 1ms`

That's just means all the available hits were selected and then sorted by the Algorithm. I don't see anything weird with it, just the "transparency" we initially wanted.

On top on that, relying on `nbHits` can be dangerous as this value can sometimes, be interpolated by the engine and thus being volatile and incorrect. See `exhaustiveNbHits` field in the response.

To finish, one rule of thumb is `relevancyStrictness=0` means that the algorithm is disabled and that no results selection/deletion is being done. It's *almost* (not exactly) equivalent to a hard sort. When `relevancyStrictness> 0`  the algorithm is enabled and it does select/remove results.

Hope that helps!

Reply:
I think the button should be displayed, so for the context of SmartSort, isSmartSorted is true as long as appliedRelevancyStrictness is not 0, but in the stats widget I think it doesn't make sense to show "x relevant results shown out of x in yms", but it should be "x results found in yms", which correlates with isSmartSorted: false

Reply:
@Haroenv  agree with what you said, except that two `isSmartSorted` variables in connectSmartSort and connectStats will have different definitions (but the same name), so confusing.
So I renamed `isSmartSorted` to `areHitsSorted` in the context of stats, and added the condition to check `nbSortedHits !== nbHits`.
https://github.com/algolia/instantsearch.js/pull/4649/commits/3c24187e10bcf4907077203cb9233fa229b383a7